File: telegram-bot-bumblebee-core/src/main/java/com/github/bumblebee/command/statistics/entity/Statistic.java
package com.github.bumblebee.command.statistics.entity;


import javax.persistence.*;
import java.time.LocalDate;

@Entity
@Table(name = "BB_STATISTICS")
public class Statistic {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private LocalDate postedDate;

    private MessageTypes messageType;

    private Integer messageCount;

    private Long chatId;

    private Long authorId;

    private String authorName;

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public LocalDate getPostedDate() {
        return postedDate;
    }

    public void setPostedDate(LocalDate postedDate) {
        this.postedDate = postedDate;
    }

    public MessageTypes getMessageType() {
        return messageType;
    }

    public void setMessageType(MessageTypes messageType) {
        this.messageType = messageType;
    }

    public Integer getMessageCount() {
        return messageCount;
    }

    public void setMessageCount(Integer messageCount) {
        this.messageCount = messageCount;
    }

    public Long getChatId() {
        return chatId;
    }

    public void setChatId(Long chatId) {
        this.chatId = chatId;
    }

    public Long getAuthorId() {
        return authorId;
    }

    public void setAuthorId(Long authorId) {
        this.authorId = authorId;
    }

    public String getAuthorName() {
        return authorName;
    }

    public void setAuthorName(String authorName) {
        this.authorName = authorName;
    }

    enum MessageTypes {
        TEXT,
        PICTURE
    }
}

File: telegram-bot-bumblebee-core/src/main/java/com/github/bumblebee/command/statistics/entity/Statistic.kt
package com.github.bumblebee.command.statistics.entity

import java.time.LocalDate
import javax.persistence.*

@Entity
@Table(name = "BB_STATISTICS")
class Statistic {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long? = null

    var postedDate: LocalDate? = null

    var messageCount: Int = 0

    var chatId: Long = 0

    var authorId: Long = 0

    var authorName: String? = null

    @Suppress("unused")
    constructor()

    constructor(postedDate: LocalDate,
                messageCount: Int,
                chatId: Long,
                authorId: Long,
                authorName: String?) {
        this.postedDate = postedDate
        this.messageCount = messageCount
        this.chatId = chatId
        this.authorId = authorId
        this.authorName = authorName
    }
}